Makefile

```diff
@@ -1,51 +1,77 @@
+LDFLAGS += -L./lib -Wl,-rpath,$(LIBDIR) -Wl,--whole-archive -lpthread -Wl,--no-whole-archive
+
+BAMTOOLS_LIB_PREFIX = lrez_
+BAMTOOLS_LIB = lib/lib$(BAMTOOLS_LIB_PREFIX)bamtools$(SHLIB_EXT)
 
-LREZ_INC = $(curDir)/src/include/
-LREZ_LIB = $(curDir)/lib/
+BAMTOOLS_INC = ./include/bamtools/
+LREZ_INC = ./src/include/
 
-LDFLAGS_BAMTOOLS = -lbamtools -L$(BAMTOOLS_LIB)
-LDFLAGS_LREZ = -llrez -L$(LREZ_LIB)
-LDFLAGS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_LREZ = -llrez
+LIBS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_BAMTOOLS = -l$(BAMTOOLS_LIB_PREFIX)bamtools
 
 MAIN = src/main.o
 REVCOMP = src/reverseComplement.o
 SOURCE = src/alignmentsRetrieval.o src/barcodesComparison.o src/barcodesExtraction.o src/indexManagementBam.o src/utils.o src/indexManagementFastq.o src/readsRetrieval.o src/gzIndex.o
 SUBCOMMANDS = src/subcommands/compare.o src/subcommands/extract.o src/subcommands/help.o src/subcommands/indexBam.o src/subcommands/queryBam.o src/subcommands/indexFastq.o src/subcommands/queryFastq.o
 
 EXEC = bin/LRez
-LIB = lib/liblrez.${SHLIB_EXT}
-
-all: directories $(LIB) $(EXEC)
+LIB = lib/liblrez${SHLIB_EXT}
 
+all: $(LIB) $(EXEC)
 
 directories:
 	mkdir -p bin/ lib/
 
-$(LIB): $(SUBCOMMANDS) $(SOURCE) $(REVCOMP)
-	$(CC) -fPIC -shared -o $(LIB) $(SUBCOMMANDS) $(SOURCE) $(REVCOMP) $(LDFLAGS_BAMTOOLS) -Wl,-rpath,$(BAMTOOLS_LIB) $(LDFLAGS_BOOST_LZ_LM)
+$(BAMTOOLS_LIB):
+	mkdir -p bamtools/build
+	cd bamtools/build && \
+		cmake \
+			-DBUILD_SHARED_LIBS=ON \
+			-DCMAKE_SHARED_LIBRARY_PREFIX_CXX=lib$(BAMTOOLS_LIB_PREFIX) \
+			-DCMAKE_INSTALL_LIBDIR=lib \
+			-DCMAKE_INSTALL_PREFIX=$(curDir) \
+			.. && \
+		$(MAKE) -C src/api && \
+		cmake -DCMAKE_INSTALL_LOCAL_ONLY=1 -P src/cmake_install.cmake && \
+		cmake -P src/api/cmake_install.cmake
+
+$(LIB): $(SUBCOMMANDS) $(SOURCE) $(REVCOMP) $(BAMTOOLS_LIB) directories
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -o $(LIB) $(SUBCOMMANDS) $(SOURCE) $(REVCOMP) $(LIBS_BAMTOOLS) $(LIBS_BOOST_LZ_LM)
 
-$(EXEC): $(MAIN) $(LIB)
-	$(CC) -o $(EXEC) $(MAIN) $(LDFLAGS_LREZ) -Wl,-rpath,$(LREZ_LIB) $(LDFLAGS_BAMTOOLS) -Wl,-rpath,$(BAMTOOLS_LIB) $(LDFLAGS_BOOST_LZ_LM)
+$(EXEC): $(MAIN) $(LIB) directories
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $(EXEC) $(MAIN) $(LIBS_LREZ) $(LIBS_BAMTOOLS) $(LIBS_BOOST_LZ_LM)
 
-src/%.o: src/%.cpp
-	$(CC) -o $@ -c $< $(CFLAGS) -I$(BAMTOOLS_INC) -I$(LREZ_INC)
+src/%.o: src/%.cpp $(BAMTOOLS_LIB)
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -I$(BAMTOOLS_INC) -I$(LREZ_INC) -o $@ -c $<
 
 src/reverseComplement.o: src/reverseComplement.cpp
-	$(CC) -o src/reverseComplement.o -c src/reverseComplement.cpp $(CFLAGS) -I$(LREZ_INC)
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -I$(LREZ_INC) -o src/reverseComplement.o -c src/reverseComplement.cpp
+
 
+install: $(EXEC) $(LIB)
+	mkdir -p $(DESTDIR)$(BINDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	cp -a $(EXEC) $(DESTDIR)$(BINDIR)/
+	cp -a $(LIB)* $(DESTDIR)$(LIBDIR)/
+	cp -a $(BAMTOOLS_LIB)* $(DESTDIR)$(LIBDIR)/
```

Comment:
`make install` support :). Overall, the Conda recipe's build script can be simplified to two-liner with all changes:
```sh
make -j"${CPU_COUNT}"
make install
```

Reply:
The `DESTDIR` thing is nothing we use with `conda-build` because we always directly install into the final prefix. But other packaging systems use `DESTDIR` to copy the files to some place that can simply be `tar`'ed to create a package the user can install/extract.